openfisca_core/taxbenefitsystems.py

```diff
@@ -137,7 +137,7 @@ def add_variables_from_file(self, file_path):
         """
         try:
             file_name = path.splitext(path.basename(file_path))[0]
-            module_name = '{}_{}'.format(id(self), file_name)  # If two tax and benefit systems load the same module, the second one should not replace the first one. Hence this unique module name.
+            module_name = '{}_{}_{}'.format(id(self), hash(path.abspath(file_path)), file_name)  # If two tax and benefit systems load the same module, the second one should not replace the first one. Hence this unique module name.
```

Comment:
Can you add a comment that justify such a definition for `module_name`?

Indeed, previously the `module_name` was only match with `endsWith` but you updated this with `==`.
My quick online investigation didn't provide me with the format of `__module__`.
Format that you explicit here with `module_name = '{}_{}_{}'.format(id(self), hash(path.abspath(file_path)), file_name)`

Reply:
In the regular python world, `__module__` looks like `'openfisca_france.model.revenus.activite.salarie'`, so our name is not standard.

We are doing something quite specific when loading the legislation. We dynamically load many variables files as python modules. In this context, we don't know what the standard module name would be. There may even not be one: we can load _any_ file, including files that would not be accessible via `from ... import ...`, because they are not in an installed package.

We thus need to choose a name for the module we import.

As Python remembers all loaded modules by their name, we need to take some precautions to avoid collisions:

- If we load two different files with the same file name, they must have different module names. This is the issue you reported. Hence the absolute file path hash in the module name.
- If we load the same files for two different tax and benefit system, we also want them to be separate modules, for more tricky reasons. This is #612. Hence the `id(self)` in the module name.

The `endsWith` did not make a lot of sense in the context, I think it was a remaining piece of a previous implementation.

Reply:
OK, thanks for the clarification (it is matching because of `load_module(module_name,…)` where `module_name` is of our own format.)